### PR TITLE
KAFKA-8984: Improve tagged fields documentation

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: clients/src/main/resources/common/message/README.md [hunk 1]
```diff
@@ -104,6 +107,39 @@ If a field is declared as non-nullable, and it is present in the message
 version you are using, you should set it to a non-null value before serializing
 the message.  Otherwise, you will get a runtime error.
 
+Tagged Fields
+-------------
+Tagged fields are an extension to the Kafka protocol which allows optional data
+to be attached to messages.  Tagged fields can appear at the root level of
```
Comment (soondenana) [inline clients/src/main/resources/common/message/README.md:113]: It seems like each period is followed by two spaces. Is it something to do with markdown format?

File: tests/bin/flatten_html.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    [[ -f "${CUR_FILE}" ]] || die "Unable to open input file ${CUR_FILE}"
+    while IFS= read -r LINE; do
+        if [[ $LINE =~ \#include\ virtual=\"(.*)\" ]]; then
+            local INCLUDED_FILE="${BASH_REMATCH[1]}"
+            if [[ $INCLUDED_FILE =~ ../includes/ ]]; then
+                : # ignore ../includes
+            else
+                pushd "$(dirname "${CUR_FILE}")" &> /dev/null \
+                    || die "failed to change directory to directory of ${CUR_FILE}"
+                process_file "${INCLUDED_FILE}"
```
Comment (soondenana) [inline tests/bin/flatten_html.sh:57]: Ah ha!! Recursion in bash. Hopefully no one will create a circular include :-)

File: tests/bin/flatten_html.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    case $arg in
+        f) FILE=$OPTARG;;
+        h) usage; exit 0;;
+        *) echo "Error parsing command-line arguments."
+            usage
+            exit 1;;
+    esac
+done
+
+[[ -z "${FILE}" ]] && die "You must specify which file to process.  -h for help."
```
Comment (soondenana) [inline tests/bin/flatten_html.sh:77]: Hmm, two spaces after full stop here too (before `-h`). Is it github review that is showing it this way?

**Review by soondenana — APPROVED** [review]
Thanks for updating docs. Left a question/nit.

File: clients/src/main/resources/common/message/README.md (hunk 1 above)
Comment (cmccabe) [reply clients/src/main/resources/common/message/README.md:113]: This is consistent with the other sentences in the file.  Markdown format doesn't require this.

File: tests/bin/flatten_html.sh (hunk 2 above)
Comment (cmccabe) [reply tests/bin/flatten_html.sh:57]: I think a circular include would probably break whatever is processing this in order to display it on the server side.  After all, it has to be flattened into a single HTML document before it is shown to the user.  :)

If we did have a circular include, bash would recurse infinitely, causing it to run out of stack space and segfault after a while.

File: tests/bin/flatten_html.sh (hunk 3 above)
Comment (cmccabe) [reply tests/bin/flatten_html.sh:77]: No, it's just using two spaces after the period

**Review by hachikuji — APPROVED** [review]
LGTM

**Pull request closed without merging**